sequence_processing_pipeline/FastQCJob.py

```diff
@@ -36,35 +36,46 @@ def __init__(self, run_dir, output_directory, nprocs, nthreads,
         self.products_dir = join(self.run_dir, self.run_id)
         # for CI, create this directory if it doesn't exist already.
         self._directory_check(self.products_dir, create=True)
+
         self.jmem = jmem
         self.pool_size = pool_size
 
+        self.project_names = []
         self.commands = self._get_commands()
         split_count = self._generate_split_count(len(self.commands))
         lines_per_split = int((len(self.commands) + split_count - 1) /
                               split_count)
         self._generate_trim_files(self.commands, lines_per_split)
         self.script_path = self._generate_job_script()
 
+        for project_name in self.project_names:
+            p_path = partial(join, self.products_dir, 'FastQC', project_name)
+            makedirs(p_path('bclconvert'), exist_ok=True)
+            makedirs(p_path('filtered_sequences'), exist_ok=True)
+
     def _get_commands(self):
         results = []
 
         # gather the parameters for processing all relevant raw fastq files.
         params = self._scan_fastq_files(True)
 
-        for number_of_threads, file_path, output_path in params:
-            command = ['fastqc', '--noextract', '-t', str(number_of_threads),
-                       file_path, '-o', output_path]
+        for thread_count, fwd_file_path, rev_file_path, output_path in params:
```

Comment:
`thread_count` is basically `self.nthreads` so I'm not sure why (1) `_scan_fastq_files` (`params`) needs to store it too and (2) why is it needed in a  per fastq pairs. In other words, I think it should be fine to remove it from `_scan_fastq_files`  and just use  `self.nthreads` here.